sympy/ntheory/generate.py

```diff
@@ -217,11 +284,12 @@ def primerange(self, a, b=None):
             b = _as_int_ceiling(b)
         if a >= b:
             return
-        self.extend(b)
-        yield from self._list[bisect_left(self._list, a):
-                              bisect_left(self._list, b)]
+        _list = self._extend(b)
+        ai = bisect_left(_list, a)
+        bi = bisect_left(_list, b)
+        yield from _list[ai:bi]
```

Comment:
The `_list` is append-only, right? So the fact that `_list` is possibly mutable during the iteration isn't a problem because there will only ever be new values appended after `bi`.

Reply:
The `self._list` can be reset by the `_reset` method but it can also be made shorter by a parallel thread assigning a shorter list, basically the same as https://github.com/Quansight-Labs/free-threaded-compatibility/pull/302#discussion_r2632771021. We cannot depend on `self._list` being anything after our single atomic `_list = self._list` read apart from:

- It has length at least 6
- It holds valid data for its length.
- It is an object that is never mutated in place.

These are not sufficient to know that `self._list[ai:bi]` gives the slice we want so we need our local `_list`.